feat: support text_mode collapse recursive

CahierX opened the pull request:
To solve https://github.com/josdejong/svelte-jsoneditor/issues/557 issues
Add text_mode collapse recursive

josdejong commented:
That is impressive, this looks promising! 😎 

The code looks really neat. Whilst testing I came across two issues though:

1. When I collapse the whole document with `refJsonEditor.collapse([], true)`, the recursiveness doesn't work: only the root object or array is collapsed.
2. When I open a large document and I perform `refJsonEditor.collapse([], true)`, only the first part of the items in an array is collapsed. For example take [the "10k users" example from here](https://jsoneditoronline.org/indepth/datasets/json-file-example/#a-large-json-document). This may be related to CodeMirror only rendering the visible part of the contents, I'm not sure.

I think issue (1) is likely a small bug but (2) may be tricky.

CahierX commented:
> That is impressive, this looks promising! 😎
> 
> The code looks really neat. Whilst testing I came across two issues though:
> 
> 1. When I collapse the whole document with `refJsonEditor.collapse([], true)`, the recursiveness doesn't work: only the root object or array is collapsed.
> 2. When I open a large document and I perform `refJsonEditor.collapse([], true)`, only the first part of the items in an array is collapsed. For example take [the "10k users" example from here](https://jsoneditoronline.org/indepth/datasets/json-file-example/#a-large-json-document). This may be related to CodeMirror only rendering the visible part of the contents, I'm not sure.
> 
> I think issue (1) is likely a small bug but (2) may be tricky.

I will try to fix it

CahierX commented:
I also submitted a [PR](https://github.com/codemirror/language/pull/8) to CodeMirror. If it gets merged, we’ll be able to remove our `createCustomFoldService`.

CahierX commented:
Sorry for the delay in submitting the code - I've been quite busy with work lately.

Regarding the performance issue with `codeMirrorView.dispatch({effects})` when handling large JSON files, I investigated the CodeMirror source code but couldn't find any viable directions for performance optimization. Therefore, I've implemented solution 2 to address this issue.

I've added a folding progress bar with cancellation support.
<img width="728" height="562" alt="Clipboard_Screenshot_1755008872" src="https://github.com/user-attachments/assets/385a2a05-3fe2-40f7-9360-26a7fc3a2c67" />

josdejong commented:
Thanks, awesome job!

> Are you feeling better now?

Yes 😅. It was just the flu, nothing special though quite annoying.

> As for the points you mentioned (1, 2, and 3), I think they’re all very reasonable. I’ll try to add them in future PRs when I have the time.

👍

CahierX commented:
> Just for reference: I've added "Expand All" and "Collapse All" buttons to the menu in text mode now, see #563. I love it, thanks again for making this possible @CahierX!

🎉🎉
